From 0f6f918235f91a016eb47bf600d6aab2cbec7f16 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Mon, 1 Feb 2016 13:05:21 +0800
Subject: [PATCH 0912/5242] MLK-12358: ARM: dts: imx7d-val-m4: disable sim1
 and reg_can2_3v3

commit  ea15a3ea9bae972b2eeef8427969363787235e67 from
https://source.codeaurora.org/external/imx/linux-imx.git

- Disable the sim1 on imx7d val board, because that there
are pin conflictions between sim1 and flexcan2 used by M4.
- Disable can2 is not enough, since the reg_can2_3v3 will be
turned off by the regulator framework and that will impact
can2 in m4 side even if can2 driver disabled in A7 side.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Tested-by: Ge Wang <G.Wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
index 65821e6..e06a53f 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
@@ -51,10 +51,18 @@
 	reg = <0x00901000 0xf000>;
 };
 
+&reg_can2_3v3 {
+	status = "disabled";
+};
+
 &rpmsg{
 	status = "okay";
 };
 
+&sim1 {
+	status = "disabled";
+};
+
 &uart2 {
 	status = "disabled";
 };
-- 
1.7.9.5

